monitor/uidmonitor/processor.go

```diff
@@ -0,0 +1,346 @@
+	metadataExtractor rpcmonitor.RPCMetadataExtractor,
+	releasePath string) *UIDProcessor {
+	if puHandler == nil {
+		zap.L().Error("PuHandler cannot be nil")
+		return nil
+	}
+	return &UIDProcessor{
+		collector:         collector,
+		puHandler:         puHandler,
+		metadataExtractor: metadataExtractor,
+		netcls:            cgnetcls.NewCgroupNetController(releasePath),
+		contextStore:      contextstore.NewContextStore(storePath),
+		storePath:         storePath,
+		regStart:          regexp.MustCompile("^[a-zA-Z0-9_].{0,11}$"),
+		regStop:           regexp.MustCompile("^/trireme/[a-zA-Z0-9_].{0,11}$"),
+		putoPidMap:        cache.NewCache(),
+		pidToPU:           cache.NewCache(),
+	}
+}
+
+// NewUIDProcessor creates a default Linux processor with the standard trireme path
+func NewUIDProcessor(collector collector.EventCollector,
+	puHandler monitor.ProcessingUnitsHandler,
+	metadataExtractor rpcmonitor.RPCMetadataExtractor,
+	releasePath string) *UIDProcessor {
+	return NewCustomUIDProcessor("/var/run/trireme/linux", collector, puHandler, metadataExtractor, releasePath)
+}
+
+// Create handles create events
+func (s *UIDProcessor) Create(eventInfo *rpcmonitor.EventInfo) error {
+
+	return s.puHandler.HandlePUEvent(eventInfo.PUID, monitor.EventCreate)
+}
+
+// Start handles start events
+func (s *UIDProcessor) Start(eventInfo *rpcmonitor.EventInfo) error {
+	s.Lock()
+	defer s.Unlock()
+	contextID := eventInfo.PUID
+	pids, err := s.putoPidMap.Get(contextID)
+	var runtimeInfo *policy.PURuntime
+	if err != nil {
+		runtimeInfo, err = s.metadataExtractor(eventInfo)
+		if err != nil {
+			return err
+		}
+
+		publishedContextID := contextID + runtimeInfo.Options().CgroupMark
+		// Setup the run time
+		if err = s.puHandler.SetPURuntime(publishedContextID, runtimeInfo); err != nil {
+			return err
+		}
+
+		defaultIP, _ := runtimeInfo.DefaultIPAddress()
+		if perr := s.puHandler.HandlePUEvent(publishedContextID, monitor.EventStart); perr != nil {
+			zap.L().Error("Failed to activate process", zap.Error(perr))
+			return perr
+		}
+
+		if err = s.processLinuxServiceStart(eventInfo, runtimeInfo); err != nil {
+			zap.L().Error("processLinuxServiceStart", zap.Error(err))
+			return err
+		}
+
+		s.collector.CollectContainerEvent(&collector.ContainerRecord{
+			ContextID: contextID,
+			IPAddress: defaultIP,
+			Tags:      runtimeInfo.Tags(),
+			Event:     collector.ContainerStart,
+		})
+		entry := &puToPidEntry{
+			Info:               runtimeInfo,
+			publishedContextID: publishedContextID,
+			pidlist:            map[string]bool{},
+		}
+
+		entry.pidlist[eventInfo.PID] = true
+		s.putoPidMap.Add(contextID, entry)
+		s.pidToPU.Add(eventInfo.PID, contextID)
+		// Store the state in the context store for future access
+		return s.contextStore.StoreContext(contextID, &StoredContext{
+			EventInfo: eventInfo,
+			MarkVal:   runtimeInfo.Options().CgroupMark,
+		})
+
+	}
+
+	pids.(*puToPidEntry).pidlist[eventInfo.PID] = true
```

Comment:
extremely dangerous, plus I reiterate, use a `map[string]struct{}` not `map[string]bool`